pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -115,28 +122,62 @@ func (a *AuthorizedEntryFetcherWithEventsBasedCache) updateRegistrationEntriesCa
 
 	seenMap := map[string]struct{}{}
 	for _, event := range resp.Events {
+		// If there is a gap in the event log the missed events for later processing
+		if event.EventID != a.lastRegistrationEntryEventID+1 {
```

Comment:
A small bug exists in this logic.

upon the first pull of RegistrationEntryEvents, a.lastRegistrationEntryEventID will be initialized to zero, which means that a.missedREgistrationEntryEvents would be populated by all event IDs between zero and the first read RegistrationEntryEventID.  

What is needed is a boolean flag, that indicates the first RegistrationEntryEventID was read, so we don't include ranges of skipped RegistrationEntryEventIDs that haven't been skipped.

Reply:
Updated to check for zero